src/Events/Received/Plugin/AbstractVisibilityEvent.ts

```diff
@@ -0,0 +1,19 @@
+import AbstractStateEvent from '@/Events/Received/Plugin/AbstractStateEvent';
+import { ControllerType } from '@/Events/Received/Plugin/ControllerType';
+import { BaseVisibilityType } from '@/StreamdeckTypes/Sent/BaseVisibilityType';
+
+export default abstract class AbstractVisibilityEvent extends AbstractStateEvent {
+  protected readonly eventPayload!: BaseVisibilityType;
+
+  /**
+   * Return the controller type associated with this specific event. Used to determine if a specific event is on a
+   * standard button or a rotary dial. This will not change during an action's lifecycle.
+   * For plugins running on Stream Deck software v5, this will always return "Keypad".
+   * @returns {ControllerType} The controller type (either "Keypad" or "Encoder") of this action's event.
+   */
+  public get controller(): ControllerType {
```

Comment:
only just thought about how this will be used by plugins. do they need to import the enum then?
maybe making ControllerType = 'Keypad' | 'Encoder' would be better in that case.

Reply:
did some testing, and it seems it's fine this way, as it can still be used with just strings - so probably no import of the enum necessary 👍